applications/vanilla/controllers/class.postcontroller.php

```diff
@@ -162,6 +162,10 @@ public function discussion($CategoryUrlCode = '') {
             $this->title(t('New Discussion'));
         }
 
+        if (isset($this->Draft) && $this->Draft->Type) {
```

Comment:
That would make the `touchValue('Type', $this->Data, 'Discussion');` line not set the type to `Discussion` if it's set to null since the array key exists.